### PR TITLE
create and setup etcd EBS volume

### PR DESCRIPTION
**Pull request by fgimenez**
Towards https://github.com/giantswarm/giantswarm/issues/2023 https://github.com/giantswarm/aws-operator/issues/508

First tests worked but still not verified that the volume was properly mounted and used.

### CODE REVIEW

File: service/templates/cloudformation/guest/instance.yaml [hunk 1]
```diff
@@ -16,4 +16,16 @@
...
+    Properties:
+      Size: 100
+      VolumeType: gp2
+      AvailabilityZone: !GetAtt MasterInstance.AvailabilityZone
+  MountPoint:
+    Type: AWS::EC2::VolumeAttachment
+    Properties:
+      InstanceId: !Ref MasterInstance
+      VolumeId: !Ref EtcdVolume
+      Device: /dev/sdh
```
Comment (rossf7) [inline service/templates/cloudformation/guest/instance.yaml]: Could you add a name tag to the volume?

Comment (fgimenez) [reply service/templates/cloudformation/guest/instance.yaml]: Sure done

**Comment by fgimenez** [comment]
@rossf7 all good now with the mount unit, pls take another look. 

I've taken the opportunity to add an e2e test that reboots the master instance and checks that the guest is ready (API up and nodes running) after a while, a successful log now looks like this https://circleci.com/gh/giantswarm/aws-operator/2681. This is actually the first e2e test (what we had so far was just test preparation) and IMO can be useful not only for stating that the etcd volume is properly mounted after reboot, let me know WDYT.

**Comment by fgimenez** [comment]
Etcd volume is still not mounting properly, we need a unit to format the volume before mounting. Now extending the test to uncover the failure and then adding the code to fix it.

File: integration/operator_test.go [hunk 2]
```diff
@@ -115,7 +119,54 @@ func TestMain(m *testing.M) {
...
+			masterID,
+		},
+	}
+	_, err = c.EC2.RebootInstances(rebootInput)
+	if err != nil {
+		t.Errorf("unexpected error rebooting  master %v", err)
+	}
+
+	if err := f.WaitForAPIDown(); err != nil {
+		t.Errorf("unexpected error wainting for master shutting down %v", err)
```
Comment (rossf7) [inline integration/operator_test.go]: typo on waiting

File: integration/operator_test.go [hunk 3]
```diff
@@ -115,7 +119,54 @@ func TestMain(m *testing.M) {
...
+	}
+	res, err := c.EC2.DescribeInstances(describeInput)
+	if err != nil {
+		t.Errorf("unexpected error getting master id %v", err)
+	}
+	if len(res.Reservations) != 1 {
+		t.Errorf("unexpected number of reservations %d", len(res.Reservations))
+	}
+	if len(res.Reservations[0].Instances) != 1 {
+		t.Errorf("unexpected number of reservations %d", len(res.Reservations[0].Instances))
```
Comment (rossf7) [inline integration/operator_test.go]: nitpick but should error text say instances?

**Comment by fgimenez** [comment]
All working fine with latest changes, the state is kept between reboots.

**Pull request merged**